public/app/features/dashboard/state/initDashboard.ts

```diff
@@ -76,10 +76,14 @@ async function fetchDashboard(
         return dashDTO;
       }
       case DashboardRoutes.Public: {
-        return await dashboardLoaderSrv.loadDashboard('public', args.urlSlug, args.accessToken);
+        return await dashboardLoaderSrv.loadDashboard('public', args.urlSlug ?? '', args.accessToken ?? '');
       }
       case DashboardRoutes.Normal: {
-        const dashDTO: DashboardDTO = await dashboardLoaderSrv.loadDashboard(args.urlType, args.urlSlug, args.urlUid);
+        const dashDTO: DashboardDTO = await dashboardLoaderSrv.loadDashboard(
+          args.urlType,
+          args.urlSlug ?? '',
+          args.urlUid ?? ''
+        );
```

Comment:
I'm not a fan of this change - I don't think empty-string should be used as a stand in for "empty value that satisfies non-nullable argument".

Also I think this is a non-trivial change to behaviour, "potentially" now trying to request different API resources than before when the values are undefined.

I think the better solution is to:
 - accurately type `loadDashboard` with nullable slug and uid arguments (?)
 - unpack and narrow this `args` object (urlType, urlSlug, urlUid, accessToken) and return error/"unknown route" when they're missing in the appropriate branch.

I can understand if you don't want to do that immediately in this PR...

Reply:
@joshhunt attempted to handle this with https://github.com/grafana/grafana/pull/69860/commits/a5ed2e7f246fecd5f6eaf378c82696a35965db97

wdyt? 🤔

Reply:
👍🏻  Will need @grafana/dashboards-squad to peek at this though.